lifecycle.md

```diff
@@ -48,9 +48,24 @@ Note: this content should be moved to another location as it is not lifecycle re
 
 Build and test images are referenced in repo infra files, across a variety of `main` and `release/*` branches. Updating these references is a multi-step detail-oriented task. It is a pain, but necessary.
 
-We use version specific references in infra to ensure that our CI builds are reliable. One can imagine using floating OS tags (such as `debian-oldest` and `debian-latest`), however such an approach would be guaranteed to break our build. We know that since we often see build and test breaks that need addressing in PRs where we update build and test images.
+Several types of tag styles are provided:
+* version-specific: The tag name includes the distro's version. Example: `alpine-3.21-helix-amd64`
+* floating: The tag name has no version. Instead it is routinely updated to reference a new version as the distro's lifecycle progresses. There are two sub-types of floating tags:
+  * latest: References the latest validated version of the distro. Example: `alpine-latest-helix-amd64`
+  * oldest: References the oldest supported version of the distro. Example: `alpine-oldest-helix-amd64`
 
-At times, it may be necessary to use a [fixed image reference](https://github.com/dotnet/runtime/pull/110199#discussion_r1859075989) for build reliability. If this is ever done, a tracking issue should be created (before the PR is merged) so that we remember to resolve the underlying issue and update the image reference.
+FLoating tags are beneficial for repos that are not susceptible to breaking changes that occur from new distro versions because the source that references the tag doesn't need to be updated in order to make use of the new version.
+Conversely, some repos may be susceptible to distro breaking changes in which case the version-specific tags should be used ([dotnet/runtime](https://github.com/dotnet/runtime) is an example of such a repo).
+
+There is a vetting process before a `latest` floating tag gets moved to a newer distro version:
+
+1. Version-specific tags for the new distro version are provided.
+1. The [dotnet/runtime](https://github.com/dotnet/runtime) repo is updated to reference the new version-specific tags.
```

Comment:
Aren't there going to be cases where images aren't referenced by the runtime?

I don't like this vetting process in a way.  If vetting is required you shouldn't be using the version specific tags.  I tend to learn towards the .NET Docker Alpine floating tag policy.  The new distro is made available one month, the next month the floating tag is updated.  This is a more scalable model.

Reply:
That works as a policy. If we found an issue in runtime, we'd pause roll-out in dotnet-docker. We'd do the same with this system. It does mean that we'll have to be intentional w/ensuring we get coverage in runtime pretty quickly.